### ISSUE TITLE
particle crash viscoelastic_bending_beam_particles.prm

### ISSUE THREAD

**Issue by tjhei** [geodynamics/aspect#4067]
with latest deal.ii master:
```
--------------------------------------------------------
An error occurred in line <131> of file <../source/particles/property_pool.cc> in function
    void dealii::Particles::PropertyPool<dim, spacedim>::deregister_particle(dealii::Particles::PropertyPool<dim, spacedim>::Handle&) [with int dim = 2; int spacedim = 2; dealii::Particles::PropertyPool<dim, spacedim>::Handle = unsigned int]
The violated condition was: 
    currently_available_handles.size() < locations.size()
Additional information: 
    Trying to deallocate a particle when none are allocated. This can
    happen if all handles were deallocated already before calling this
    function.

Stacktrace:
-----------
#0  /home/heister/deal-git/build-64/lib/libdeal_II.g.so.10.0.0-pre: dealii::Particles::PropertyPool<2, 2>::deregister_particle(unsigned int&)
#1  /home/heister/deal-git/build-64/lib/libdeal_II.g.so.10.0.0-pre: dealii::Particles::ParticleHandler<2, 2>::clear_particles()
#2  /home/heister/deal-git/build-64/lib/libdeal_II.g.so.10.0.0-pre: dealii::Particles::ParticleHandler<2, 2>::~ParticleHandler()
#3  /home/heister/deal-git/build-64/lib/libdeal_II.g.so.10.0.0-pre: dealii::Particles::ParticleHandler<2, 2>::~ParticleHandler()
#4  /home/heister/aspect-git/build64/aspect: std::default_delete<dealii::Particles::ParticleHandler<2, 2> >::operator()(dealii::Particles::ParticleHandler<2, 2>*) const
#5  /home/heister/aspect-git/build64/aspect: std::unique_ptr<dealii::Particles::ParticleHandler<2, 2>, std::default_delete<dealii::Particles::ParticleHandler<2, 2> > >::~unique_ptr()
#6  /home/heister/aspect-git/build64/aspect: aspect::Particle::World<2>::~World()
#7  /home/heister/aspect-git/build64/aspect: aspect::Particle::World<2>::~World()
#8  /home/heister/aspect-git/build64/aspect: std::default_delete<aspect::Particle::World<2> >::operator()(aspect::Particle::World<2>*) const
#9  /home/heister/aspect-git/build64/aspect: std::unique_ptr<aspect::Particle::World<2>, std::default_delete<aspect::Particle::World<2> > >::reset(aspect::Particle::World<2>*)
#10  /home/heister/aspect-git/build64/aspect: aspect::Simulator<2>::~Simulator()
#11  /home/heister/aspect-git/build64/aspect: void run_simulator<2>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, bool)
#12  /home/heister/aspect-git/build64/aspect: main
--------------------------------------------------------

environment: line 1: 28524 Aborted                 (core dumped) /home/heister/aspect-git/build64/aspect $prm.tmp > /dev/null
run of viscoelastic_bending_beam_particles.prm failed
```

**Comment by tjhei** [comment]
also viscoelastic_stress_build-up_particles.prm

**Comment by gassmoeller** [comment]
Both of those should be hopefully fixed by #4063. I have not checked yet, but can do tomorrow.

**Comment by gassmoeller** [comment]
I think there are at least two issues, one in aspect, one in deal.II. I have preliminary fixes for them, but will need to do more testing tomorrow.
